model/minsky.cc

```diff
@@ -264,6 +265,35 @@ namespace minsky
     GroupPtr g(new Group);
     canvas.setItemFocus(model->addGroup(g));
     m.populateGroup(*g);
+    g->resizeOnContents();	  
+  }
+
+    // Fix for ticket 1080
+  void Minsky::pasteItems()
+  {
+    istringstream is(getClipboard());
+    xml_unpack_t unpacker(is);
+    schema2::Minsky m;
+    xml_unpack(unpacker, "Minsky", m);
+    GroupPtr g(new Group);
+    canvas.setItemFocus(model->addGroup(g));
+    m.populateGroup(*g);
+    if (auto p=g->group.lock())
+    {
+	   p->moveContents(*g); 
+	   canvas.item.reset();
+       model->recursiveDo(&GroupItems::items,
+                              [&](const Items&, Items::const_iterator i)
+                              {
+                                 canvas.item=*i;
+                                 canvas.item->selected=true;
+                                 canvas.selection.ensureItemInserted(canvas.item);
+                                 return false;  
+                              });       
+       canvas.setItemFocus(model->addItem(canvas.item)); 
```

Comment:
canvas.item shouldn't need to be added to model? 
What about simply calling ungroup on g, then using a simple loop to add the top level items to the selection?

Reply:
This code belongs to a different bugfix, I don't know why it appears here. I will revert this commit, pull your remote master into bugfix-1043 again, push to remote and only then commit new changes. Sorry about this.

Reply:
Is this still to be fixed?